teos/internal_api.py

```diff
@@ -144,3 +147,71 @@ def get_all_appointments(self, request, context):
         appointments.update({"watcher_appointments": watcher_appointments, "responder_trackers": responder_trackers})
 
         return GetAllAppointmentsResponse(appointments=appointments)
+
+    def get_appointments(self, request, context):
+        try:
+            Inspector.check_locator(request.locator)
+        except InspectionFailed as e:
+            context.set_details(f"Invalid locator: {e.reason}")
+            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+            return GetAppointmentsResponse()
+
+        with self.rw_lock.gen_rlock():
+            uuids = self.watcher.locator_uuid_map.get(request.locator)
```

Comment:
Also, this won't work straightaway. If the locator is not in the Watcher, it can be in the Responder.

I think there's no good way of getting al this atm, since if the appointment is in the Responder, there is no way of fetching it by locator. We may need a new map for that.

Reply:
Mostly done in 5154047f2fa083a2bedca19a687c3780f44c1d44 and 9ea435a3086f5dc25c1e1b6a46e081a492b875c3. Removed the get_appointments cli command in 3d3c47c0c6d5a45ef4f56476e09514440afd0007, will be added as a separate ticket.